plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java

```diff
@@ -1751,6 +1755,124 @@ public void testOptimizeUsingForcedPartitioning()
         assertThat(getAllDataFilesFromTableDirectory(tableName)).isEqualTo(union(initialFiles, updatedFiles));
     }
 
+    @Test
+    public void testRegisterTableWithTableLocation()
+    {
+        String tableName = "test_register_table_with_table_location_" + randomTableSuffix();
+
+        assertUpdate(format("CREATE TABLE %s (a int, b varchar, c boolean)", tableName));
+        assertUpdate(format("INSERT INTO %s values(1, 'INDIA', true)", tableName), 1);
+        assertUpdate(format("INSERT INTO %s values(2, 'USA', false)", tableName), 1);
+
+        String tableLocation = getTableLocation(tableName);
+        String showCreateTableOld = (String) computeActual("SHOW CREATE TABLE " + tableName).getOnlyValue();
+        // Drop table from metastore and use the same table to register again
+        dropTableFromMetastore(tableName);
+
+        assertUpdate("CALL system.register_table (CURRENT_SCHEMA, '" + tableName + "', '" + tableLocation + "')");
+        String showCreateTableNew = (String) computeActual("SHOW CREATE TABLE " + tableName).getOnlyValue();
+
+        assertThat(showCreateTableOld).isEqualTo(showCreateTableNew);
+        assertThat(query(format("SELECT * FROM %s", tableName)))
+                .matches("VALUES " +
+                        "ROW(INT '1', VARCHAR 'INDIA', BOOLEAN 'true'), " +
+                        "ROW(INT '2', VARCHAR 'USA', BOOLEAN 'false')");
+        assertUpdate(format("DROP TABLE %s", tableName));
+    }
+
+    @Test
+    public void testRegisterTableWithComments()
+    {
+        String tableName = "test_register_table_with_comments_" + randomTableSuffix();
+
+        assertUpdate(format("CREATE TABLE %s (a int, b varchar, c boolean)", tableName));
+        assertUpdate(format("INSERT INTO %s values(1, 'INDIA', true)", tableName), 1);
+        assertUpdate(format("INSERT INTO %s values(2, 'USA', false)", tableName), 1);
+        assertUpdate(format("COMMENT ON TABLE %s is 'my-table-comment'", tableName));
+        assertUpdate(format("COMMENT ON COLUMN %s.a is 'a-comment'", tableName));
+        assertUpdate(format("COMMENT ON COLUMN %s.b is 'b-comment'", tableName));
+        assertUpdate(format("COMMENT ON COLUMN %s.c is 'c-comment'", tableName));
```

Comment:
nit: Comment on a column is enough to me (unless there's reason we want to test in all columns).